lib/rust/lexer/definition/src/token.rs

```diff
@@ -216,6 +324,124 @@ impl Default for LineEnding {
+    RawLine,
+    /// An unclosed text line literal.
+    UnclosedLine,
+
+    // === Inline Block ===
+
+    /// A format inline block text literal.
+    FormatInlineBlock,
+    /// A raw inline block text literal.
+    RawInlineBlock,
+
+    // === Block ===
+
+    /// An interpolated text block literal.
+    FormatBlock,
+    /// A raw text block literal.
+    RawBlock,
+}
+
+impl TextStyle {
+    /// Calculate the length of the delimiters for a particular style of text literal.
+    pub fn length(self) -> usize {
+        match self {
+            TextStyle::FormatLine        => 2,
+            TextStyle::RawLine           => 2,
+            TextStyle::FormatInlineBlock => 3,
+            TextStyle::RawInlineBlock    => 3,
+            TextStyle::UnclosedLine      => 1,
+            TextStyle::FormatBlock       => 3,
+            TextStyle::RawBlock          => 3,
+        }
+    }
+
+    /// Check if the text literal is a line literal.
+    pub fn is_line_literal(self) -> bool {
+        match self {
+            TextStyle::RawLine      => true,
+            TextStyle::FormatLine   => true,
+            TextStyle::UnclosedLine => true,
+            _                       => false,
+        }
+    }
+
+    /// Check if the text literal is an inline block literal.
+    pub fn is_inline_block_literal(self) -> bool {
+        match self {
+            TextStyle::FormatInlineBlock => true,
+            TextStyle::RawInlineBlock    => true,
+            _                            => false,
+        }
+    }
+
+    /// Check if the text literal is a block literal.
+    pub fn is_block_literal(self) -> bool {
+        match self {
+            TextStyle::FormatBlock => true,
+            TextStyle::RawBlock    => true,
+            _                      => false,
+        }
+    }
+}
+
+
+
+// ===================
+// === EscapeStyle ===
+// ===================
+
+/// A description of the style of escape sequence seen.
+#[derive(Clone,Copy,Debug,Eq,PartialEq)]
+pub enum EscapeStyle {
+    /// A \xNN-style byte escape.
+    Byte,
+    /// Unicode 16-bit escape sequence.
+    U16,
+    /// Unicode 21-bit escape sequence.
+    U21,
+    /// Unicode 32-bit escape sequence.
+    U32,
+    /// A literal escape character.
+    Literal,
+    /// An invalid unicode escape.
+    InvalidUnicode,
+    /// An invalid escape.
+    Invalid,
+    /// An escape slash without any following escape.
+    Unfinished,
+}
+impl EscapeStyle {
+    /// Get the length taken up in source by the delimiters to an escape type.
+    pub fn size(self) -> usize {
+        match self {
+            EscapeStyle::Byte           => 2,
+            EscapeStyle::Invalid        => 0,
+            EscapeStyle::InvalidUnicode => 0,
+            EscapeStyle::Literal        => 1,
+            EscapeStyle::U16            => 2,
+            EscapeStyle::U21            => 4,
+            EscapeStyle::U32            => 2,
+            EscapeStyle::Unfinished     => 0,
```

Comment:
a lot of magical constants here as well. You are usign the same constants in validators, also hardcoded in several places in code there